types/estree/index.d.ts

```diff
@@ -223,14 +244,35 @@ export interface VariableDeclarator extends BaseNode {
   init?: Expression | null | undefined;
 }
 
-type Expression =
-    ThisExpression | ArrayExpression | ObjectExpression | FunctionExpression |
-    ArrowFunctionExpression | YieldExpression | Literal | UnaryExpression |
-    UpdateExpression | BinaryExpression | AssignmentExpression |
-    LogicalExpression | MemberExpression | ConditionalExpression |
-    CallExpression | NewExpression | SequenceExpression | TemplateLiteral |
-    TaggedTemplateExpression | ClassExpression | MetaProperty | Identifier |
-    AwaitExpression | ImportExpression | ChainExpression;
+export interface ExpressionMap {
+  ArrayExpression: ArrayExpression;
+  ArrowFunctionExpression: ArrowFunctionExpression;
+  AssignmentExpression: AssignmentExpression;
+  AwaitExpression: AwaitExpression;
+  BinaryExpression: BinaryExpression;
+  CallExpression: CallExpression;
+  ChainExpression: ChainExpression;
+  ClassExpression: ClassExpression;
+  ConditionalExpression: ConditionalExpression;
+  FunctionExpression: FunctionExpression;
+  Identifier: Identifier ;
+  ImportExpression: ImportExpression;
+  Literal: Literal ;
+  LogicalExpression: LogicalExpression;
+  MemberExpression: MemberExpression;
+  MetaProperty: MetaProperty;
+  NewExpression: NewExpression;
+  ObjectExpression: ObjectExpression;
+  SequenceExpression: SequenceExpression;
+  TaggedTemplateExpression: TaggedTemplateExpression;
+  TemplateLiteral: TemplateLiteral ;
+  ThisExpression: ThisExpression;
+  UnaryExpression: UnaryExpression ;
+  UpdateExpression: UpdateExpression;
+  YieldExpression: YieldExpression;
+}
+
+type Expression = ExpressionMap[keyof ExpressionMap]
```

Comment:
Should there be more maps and content types? E.g., https://github.com/estree/formal/tree/master/formal-data/typescript uses multiple versions, and each version adds onto the previous.

Also, could the linked source be used to make these types?

Reply:
I think it makes sense to add more of such maps as use cases unfold. I.e. when I also handle `flow.d.ts` this way.

Should I extract `flow.d.ts` into `estree-flow` instead? Just like `estree-jsx`.